src/System.IO.FileSystem/src/System/IO/File.cs

```diff
@@ -725,5 +728,317 @@ public static void Decrypt(String path)
+        }
+
+        private static async Task InternalWriteAllLinesAsync(TextWriter writer, IEnumerable<String> contents, CancellationToken cancellationToken)
+        {
+            Debug.Assert(writer != null);
+            Debug.Assert(contents != null);
+
+            using (writer)
+            {
+                foreach (string line in contents)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    // Note that this working depends on the fix to #8662, and cannot be ported without
+                    // either also porting that fix, or explicitly checking for null.
+                    await writer.WriteLineAsync(line).ConfigureAwait(false);
+                }
+
+                cancellationToken.ThrowIfCancellationRequested();
+                await writer.FlushAsync().ConfigureAwait(false);
+            }
+        }
+
+        private static async Task InternalWriteAllTextAsync(StreamWriter sw, string contents, CancellationToken cancellationToken)
+        {
+            using (sw)
+            {
+                if (!string.IsNullOrEmpty(contents))
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    char[] buffer = new char[DefaultBufferSize];
+                    int count = contents.Length;
+                    int index = 0;
+                    while (index < count)
+                    {
+                        cancellationToken.ThrowIfCancellationRequested();
+                        int batchSize = Math.Min(DefaultBufferSize, count);
+                        contents.CopyTo(index, buffer, 0, batchSize);
+                        await sw.WriteAsync(buffer, 0, batchSize).ConfigureAwait(false);
+                        index += batchSize;
+                    }
+
+                    cancellationToken.ThrowIfCancellationRequested();
+                    await sw.FlushAsync().ConfigureAwait(false);
+                }
+            }
+        }
+
+        public static Task AppendAllTextAsync(string path, string contents, CancellationToken cancellationToken = default(CancellationToken))
+            => AppendAllTextAsync(path, contents, UTF8NoBOM, cancellationToken);
+
+        public static Task AppendAllTextAsync(string path, string contents, Encoding encoding, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (path == null)
+                throw new ArgumentNullException(nameof(path));
+            if (encoding == null)
+                throw new ArgumentNullException(nameof(encoding));
+            if (path.Length == 0)
+                throw new ArgumentException(SR.Argument_EmptyPath, nameof(path));
+
+            return cancellationToken.IsCancellationRequested
+                ? Task.FromCanceled(cancellationToken)
+                : InternalWriteAllTextAsync(AsyncStreamWriter(path, true, encoding), contents, cancellationToken);
```

Comment:
Named parameters? Same for all of the other places with true, false, null literals being passed directly.